File: src/matchMaker.Services/User/User.BusinessLogic/DTOs/Response/UserReportDto.cs
using Common.Enums;

namespace User.BusinessLogic.DTOs.Response;

public class UserReportDto
{
    public long Id { get; set; }
    public long ReporterUserId { get; set; }
    public long ReportedUserId { get; set; }
    public string Reason { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }

    public string ReporterEmail { get; set; } = string.Empty;
    public string ReportedEmail { get; set; } = string.Empty;
    
    public string? ModeratorEmail { get; set; }
    
    public string Status { get; set; }
    public long ReportTypeId { get; set; }

    public string ReportType { get; set; } = string.Empty;

    public string? ModeratorComment { get; set; }

    public long? ReviewedByModeratorId { get; set; }

    public DateTime? ReviewedAt { get; set; }
}

File: src/ClientApp/src/app/dtos/report/UserReportDto.ts
export interface UserReportDto {
  id: number;
  reporterUserId: number;
  reportedUserId: number;
  reason: string;
  createdAt: string;

  reporterEmail: string;
  reportedEmail: string;

  moderatorEmail?: string | null;

  status: string;
  reportTypeId: number;
  reportType: string;

  moderatorComment?: string | null;
  reviewedByModeratorId?: number | null;
  reviewedAt?: string | null;
}
